drivers/modem/gsm_ppp.c

```diff
@@ -1299,13 +1306,10 @@ static int gsm_init(const struct device *dev)
 #endif	/* CONFIG_MODEM_SHELL */
 
 	gsm->context.is_automatic_oper = false;
-	gsm->gsm_data.rx_rb_buf = &gsm->gsm_rx_rb_buf[0];
-	gsm->gsm_data.rx_rb_buf_len = sizeof(gsm->gsm_rx_rb_buf);
-	gsm->gsm_data.hw_flow_control = DT_PROP(GSM_UART_NODE,
-						hw_flow_control);
 
-	ret = modem_iface_uart_init(&gsm->context.iface, &gsm->gsm_data,
-				DEVICE_DT_GET(GSM_UART_NODE));
+	ret = modem_iface_uart_init(&gsm->context.iface, &gsm->gsm_data, &gsm->gsm_rx_rb_buf[0],
```

Comment:
same as above, the understandability decreased, but I'm not sure if the incentive is high enough to group the rb_buf & uart related into their own group of structs